pkg/scheduler/factory_test.go

```diff
@@ -67,90 +66,233 @@ func TestCreate(t *testing.T) {
+					{"name" : "TestRequireZone", "argument" : {"labelsPresence" : {"labels" : ["zone"], "presence" : true}}},
+					{"name" : "TestNoFooLabel", "argument" : {"labelsPresence" : {"labels" : ["foo"], "presence" : false}}}
+				],
+				"priorities" : [
+					{"name" : "RackSpread", "weight" : 3, "argument" : {"serviceAntiAffinity" : {"label" : "rack"}}},
+					{"name" : "ZoneSpread", "weight" : 3, "argument" : {"serviceAntiAffinity" : {"label" : "zone"}}},
+					{
+						"name": "RequestedToCapacityRatioPriority",
+						"weight": 2,
+						"argument": {
+							"requestedToCapacityRatioArguments": {
+								"shape": [
+									{"utilization": 0,  "score": 0},
+									{"utilization": 50, "score": 7}
+								]
+							}
+						}
+					},
+					{"name" : "LabelPreference1", "weight" : 3, "argument" : {"labelPreference" : {"label" : "l1", "presence": true}}},
+					{"name" : "LabelPreference2", "weight" : 3, "argument" : {"labelPreference" : {"label" : "l2", "presence": false}}},
+					{"name" : "NodeAffinityPriority", "weight" : 2},
+					{"name" : "InterPodAffinityPriority", "weight" : 1}
+				]
+			}`),
+			wantPluginConfig: []schedulerapi.PluginConfig{
+				{
+					Name: nodelabel.Name,
+					Args: &schedulerapi.NodeLabelArgs{
+						PresentLabels:           []string{"zone"},
+						AbsentLabels:            []string{"foo"},
+						PresentLabelsPreference: []string{"l1"},
+						AbsentLabelsPreference:  []string{"l2"},
+					},
+				},
+				{
+					Name: serviceaffinity.Name,
+					Args: &schedulerapi.ServiceAffinityArgs{
+						AffinityLabels:               []string{"zone", "foo"},
+						AntiAffinityLabelsPreference: []string{"rack", "zone"},
+					},
+				},
+				{
+					Name: noderesources.RequestedToCapacityRatioName,
+					Args: &schedulerapi.RequestedToCapacityRatioArgs{
+						Shape: []schedulerapi.UtilizationShapePoint{
+							{Utilization: 0, Score: 0},
+							{Utilization: 50, Score: 7},
+						},
+						Resources: []schedulerapi.ResourceSpec{},
+					},
+				},
+			},
+			wantPlugin: &schedulerapi.Plugins{
+				QueueSort: &schedulerapi.PluginSet{Enabled: []schedulerapi.Plugin{{Name: "PrioritySort"}}},
+				PreFilter: &schedulerapi.PluginSet{Enabled: []schedulerapi.Plugin{{Name: "ServiceAffinity"}}},
+				Filter: &schedulerapi.PluginSet{
+					Enabled: []schedulerapi.Plugin{
+						{Name: "NodeUnschedulable"},
+						{Name: "TaintToleration"},
+						{Name: "NodeLabel"},
+						{Name: "ServiceAffinity"},
+					},
+				},
+				PostFilter: &schedulerapi.PluginSet{},
+				PreScore:   &schedulerapi.PluginSet{Enabled: []schedulerapi.Plugin{{Name: "InterPodAffinity"}}},
+				Score: &schedulerapi.PluginSet{
+					Enabled: []schedulerapi.Plugin{
+						{Name: "InterPodAffinity", Weight: 1},
+						{Name: "NodeAffinity", Weight: 2},
+						{Name: "NodeLabel", Weight: 6},
+						{Name: "RequestedToCapacityRatio", Weight: 2},
+						{Name: "ServiceAffinity", Weight: 6},
+					},
+				},
+				Reserve:  &schedulerapi.PluginSet{},
```

Comment:
Actually, wait for #95287 to merge first, so we can get rid of these lines

/hold

Reply:
OK

Reply:
It looks to be taking some time

/hold cancel